src/Core/ClassInfo.php

```diff
@@ -441,6 +441,10 @@ public static function parse_class_spec($classSpec)
             } elseif (is_array($token) && $token[0] === T_NS_SEPARATOR) {
                 $class .= $token[1];
                 $hadNamespace = true;
+            } elseif ($token === '.') {
+                // Treat service name separator as NS separator
+                $class .= '.';
```

Comment:
Yes, if you have an extension with different constructor args it's necessary to use a service name, since otherwise you'll end up with multiple forms being injected with a single singleton.